git/gitattr/tree.go

```diff
@@ -11,18 +12,31 @@ import (
 // Tree represents the .gitattributes file at one layer of the tree in a Git
 // repository.
 type Tree struct {
+	MP *MacroProcessor
+
+	SystemAttributes *Tree
+
+	UserAttributes *Tree
 	// Lines are the lines of the .gitattributes at this level of the tree.
 	Lines []Line
 	// Children are the named child directories in the repository.
 	Children map[string]*Tree
+
+	RepoAttributes *Tree
```

Comment:
I believe the `Lines` and `Children` elements of this structure are not actually accessed outside of our `gitattr` package, and so could instead have lowercase names like `lines` and `children`.

Since the new elements being added here will also not be used outside of the package, I'd suggest using lowercase names for them like `mp`, `systemAttributes`, etc., so we avoid unnecessarily exporting the fields.